lib/plugins/zero-conf/blockcypher/blockcypher.js

```diff
@@ -21,17 +24,32 @@ function authorize (account, toAddress, cryptoAtoms, cryptoCode) {
       })
 
       const confidence = account.confidenceFactor
-
+      const isRBFEnabled = account.rbf
       const url = `https://api.blockcypher.com/v1/btc/main/addrs/${toAddress}?${query}`
 
       return axios.get(url)
         .then(r => {
           const data = r.data
-          const sumTxRefs = txrefs => _.sumBy(txref => highConfidence(confidence, txref), txrefs)
-
-          const authorizedValue = sumTxRefs(data.txrefs) + sumTxRefs(data.unconfirmed_txrefs)
-
-          return cryptoAtoms.lte(authorizedValue)
+          if (isBitcoindAvailable && isRBFEnabled && data.unconfirmed_txrefs) {
+            _.map(unconfirmedTxref => {
+              promise.push(new Promise((resolve, reject) => { resolve(fetchRBF(unconfirmedTxref.tx_hash)) }))
+            }, data.unconfirmed_txrefs)
+            return Promise.all(promise)
+              .then(values => {
+                _.map(rbfInfo => {
+                  _.map(unconfirmedTxref => {
+                    if (rbfInfo.tx_hash === unconfirmedTxref.tx_hash) unconfirmedTxref.rbf = rbfInfo['bip125-replaceable']
+                  }, data.unconfirmed_txrefs)
+                }, values)
+                const sumTxRefs = txrefs => _.sumBy(txref => highConfidence(confidence, txref), txrefs)
+                const authorizedValue = sumTxRefs(data.txrefs) + sumTxRefs(data.unconfirmed_txrefs)
+                return cryptoAtoms.lte(authorizedValue)
+              })
+          } else {
+            const sumTxRefs = txrefs => _.sumBy(txref => highConfidence(confidence, txref), txrefs)
+            const authorizedValue = sumTxRefs(data.txrefs) + sumTxRefs(data.unconfirmed_txrefs)
+            return cryptoAtoms.lte(authorizedValue)
+          }
```

Comment:
Correct me if I'm wrong, but lines 44-46 and 49-51 are the exact same, and 44-46 have no relation with the other code that is being executed inside the condition

```suggestion
          if (isBitcoindAvailable && isRBFEnabled && data.unconfirmed_txrefs) {
            _.map(unconfirmedTxref => {
              promises.push(new Promise((resolve, reject) => { resolve(fetchRBF(unconfirmedTxref.tx_hash)) }))
            }, data.unconfirmed_txrefs)
            
            Promise.all(promises)
              .then(values => {
                _.map(rbfInfo => {
                  _.map(unconfirmedTxref => {
                    if (rbfInfo.tx_hash === unconfirmedTxref.tx_hash) unconfirmedTxref.rbf = rbfInfo['bip125-replaceable']
                  }, data.unconfirmed_txrefs)
                }, values)                
              })
          }
          
          const sumTxRefs = txrefs => _.sumBy(txref => highConfidence(confidence, txref), txrefs)
          const authorizedValue = sumTxRefs(data.txrefs) + sumTxRefs(data.unconfirmed_txrefs)
          return cryptoAtoms.lte(authorizedValue)
```

Reply:
yeah sumTxRefs doesn't but authorizedValue does depend on data